app/graphql/mutations/users/delete_user.rb

```diff
@@ -0,0 +1,22 @@
+module Mutations
+  module Users
+    class DeleteUser < ::Mutations::BaseMutation
+      argument :id, ID, required: true
+
+      type Types::UserType
+
+      def resolve(attributes)
+        id = attributes[:id]
+        delete_pairings(id)
+
+        user = User.find(id)
+        user.destroy
+      end
+private
+      def delete_pairings(id)
+        pairings = Pairing.where('pairer_id = ? AND pairee_id IS NULL', id)
```

Comment:
Suggestion here: might be a good place to create a `scope` on the Pairing model to do this, something like `available` or something.